### PR TITLE
[ENH[ Accept `mask_img` in `SurfaceLabelsMasker`

### PR DESCRIPTION
**Pull request by man-shu**
- Closes None relates to #4889 and #4756

Changes proposed in this pull request:

- [x] accept `mask_img`
- [ ] also return related attributes such as `region_atlas_`, `region_names_` and `region_ids_`
- [ ] add corresponding tests

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/maskers/surface_labels_masker.py [hunk 1]
```diff
@@ -47,6 +73,11 @@ class SurfaceLabelsMasker(TransformerMixin, CacheMixin, BaseEstimator):
             This value must be consistent with label values
             and image provided.
 
+    mask_img : :obj:`~nilearn.surface.SurfaceImage` object, optional
+        Mask to apply to labels_img before extracting signals. Defines the \
+        overall area of the brain to consider. The data for each \
+        hemisphere is of shape (n_vertices/2, n_regions).
```
Comment (Remi-Gau) [inline nilearn/maskers/surface_labels_masker.py:79]:
```suggestion
        hemisphere is of shape (n_vertices/2, n_regions).
```

not common but I don't think we should assume each hemisphere has the same number of vertices

**Comment by Remi-Gau** [comment]
> also return related attributes such as region_atlas_, region_names_ and region_ids_

regarding TODOs: I would wait on the one above.

we are making changes to the nifti label masker to keep track of those via a single look up table stored in an attriute as dataframe.

https://github.com/nilearn/nilearn/pull/4767#discussion_r1851614315

if you really need them make them private attribute (prefixed with an underscore)

**Comment by man-shu** [comment]
> if you really need them make them private attribute (prefixed with an underscore)

Ok let's wait then I don't think we need them for Parcellations #4756

**Comment by man-shu** [comment]
This is ready for review @Remi-Gau @bthirion

File: nilearn/maskers/surface_labels_masker.py [hunk 2]
```diff
@@ -361,29 +415,45 @@ def fit_transform(self, img, y=None):
...
+        # we will only get the data back according to the mask that was applied
+        # so if some labels were removed, we will only get the data for the
+        # remaining labels, the vertices that were masked out will be set to 0
+        if self.mask_img is not None:
+            mask_data = np.concatenate(
+                list(self.mask_img.data.parts.values()), axis=0
+            )
+            _, labels = _apply_mask(self, mask_data, self._labels_data)
+        else:
+            labels = self._labels_
```
Comment (Remi-Gau) [inline nilearn/maskers/surface_labels_masker.py:447]:
```suggestion
        labels = self._labels_
        if self.mask_img is not None:
            mask_data = np.concatenate(
                list(self.mask_img.data.parts.values()), axis=0
            )
            _, labels = _apply_mask(self, mask_data, self._labels_data) 
```

here too

File: nilearn/maskers/surface_labels_masker.py [hunk 3]
```diff
@@ -361,29 +414,44 @@ def fit_transform(self, img, y=None):
...
+            If a 2D array is provided, then it would be
+            (number of scans, number of elements).
+
 
         Returns
         -------
         :obj:`~nilearn.surface.SurfaceImage` object
             Mesh and data for both hemispheres.
         """
+        self._check_fitted()
```
Comment (man-shu) [inline nilearn/maskers/surface_labels_masker.py:435]: I think we'd be missing a test here

File: nilearn/maskers/tests/test_surface_labels_masker.py [hunk 4]
```diff
@@ -120,6 +120,39 @@ def test_surface_label_masker_transform(
...
+    surf_mask = SurfaceImage(surf_mesh(), mask_data)
+    masker = SurfaceLabelsMasker(labels_img=surf_label_img, mask_img=surf_mask)
+    masker = masker.fit()
+    n_timepoints = 5
+    with pytest.warns(
+        UserWarning,
+        match="the following labels were removed",
+    ):
+        signal = masker.transform(surf_img_2d(n_timepoints))
+    signal = masker.transform(surf_img_2d(n_timepoints))
```
Comment (bthirion) [inline nilearn/maskers/tests/test_surface_labels_masker.py:150]: to be removed ?

**Review by bthirion — COMMENTED** [review]
Looks mostly good.

File: nilearn/maskers/tests/test_surface_labels_masker.py (hunk 4 above)
Comment (man-shu) [reply nilearn/maskers/tests/test_surface_labels_masker.py:150]: oops yes

File: nilearn/maskers/surface_labels_masker.py [hunk 5]
```diff
@@ -361,29 +412,44 @@ def fit_transform(self, img, y=None):
         del y
         return self.fit().transform(img)
 
-    def inverse_transform(self, masked_img):
-        """Transform extracted signal back to surface object.
+    def inverse_transform(self, signals):
```
Comment (Remi-Gau) [inline nilearn/maskers/surface_labels_masker.py:415]: good catch: we also need to rename the parameter in inverse_transforl in SurfaceMasker

most masker either call it: X, signals or region_signals

would be a nice thing to standardize...

**Comment by man-shu** [comment]
Is this ok to merge @Remi-Gau @bthirion ?

**Comment by Remi-Gau** [comment]
> Is this ok to merge

merging
